lib/rules/require-default-props.js

```diff
@@ -141,6 +141,11 @@ module.exports = {
       switch (node.typeAnnotation.type) {
         case 'GenericTypeAnnotation':
           var annotation = resolveGenericTypeAnnotation(node.typeAnnotation);
+
+          if (annotation && annotation.id) {
+            annotation = findVariableByName(annotation.id.name);
+          }
+
           properties = annotation ? annotation.properties : [];
```

Comment:
Well, technically, my fix prevented this, but if for some reason you do multiple assignments, it would trigger the original error again:

```js
import type ImportedProps from "fake";
type NestedProps = ImportedProps;
type Props = NestedProps;
function Hello(props: Props) {
  return <div>Hello {props.name.firstname}</div>;
}
```